### ISSUE TITLE
ethereal vault modifier and wave spawner room are incompatible

### ISSUE THREAD

**Issue by KorGgenT** [Iskallia/Vault-public#1253]
**Describe the bug**
ethereal vault modifier and wave spawner room are incompatible

**To Reproduce**
1. get "ethereal" in a vault
2. activate the wave spawner room
3. observe that the wave will now kill you no matter what you do because you are trapped in this room with no way out except for the sweet release of death

**Expected behavior**
the wave spawner should do 1 of two things if some mobs can be immortal: ignore them in the count for the wave ending, or never spawn immortal mobs during the wave rush.

**Screenshots or Crash Report**
If applicable, add screenshots or crash reports to help explain your problem.
![image](https://github.com/user-attachments/assets/28a56de9-4f88-482d-887e-f3fe61f642bc)

**Info (please complete the following information):**
- World Type: server
- The_Vault Version 3.16.0.1

**Comment by joshuaepstein** [comment]
Good catch! Thank you for letting us know, should be fixed for the next version!
